ops/fe_deploy.tf

```diff
@@ -0,0 +1,229 @@
+# AWS credentials are automatically loaded from the default AWS CLI configuration
+provider "aws" {}
+
+# Fetches the data from ECR for the "frontend" Docker registry
+# Assumes that such a registry already exists
+data "aws_ecr_repository" "diy-ecr-fe" {
+  name = "frontend"
+}
+
+# Creates IAM policies with permissions suitable for running ECS tasks
+resource "aws_iam_role" "ecs_execution_role" {
+  name = "ecs_execution_role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "ecs_execution_policy" {
+  name = "ecs_execution_policy"
+  role = "${aws_iam_role.ecs_execution_role.id}"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ecr:GetAuthorizationToken",
+        "ecr:BatchCheckLayerAvailability",
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:BatchGetImage",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
+# Overly permissive security group allowing public access 
+# to all ports we care about. We will eventually need a more
+# comprehensive security setup.
+
+resource "aws_security_group" "allow_all" {
+  name        = "allow_all"
+  description = "Allow all inbound traffic"
+  vpc_id      = "${aws_vpc.main.id}"
+
+  ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    from_port   = 3000
+    to_port     = 3000
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+# Create private VPC to host our service.
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+}
+
+# Declare subnets in which our service will be hosted.
+resource "aws_subnet" "subnet_1" {
+  vpc_id            = "${aws_security_group.allow_all.vpc_id}"
+  cidr_block        = "10.0.0.0/25"
+  availability_zone = "us-east-1a"
+}
+
+resource "aws_subnet" "subnet_2" {
+  vpc_id            = "${aws_security_group.allow_all.vpc_id}"
+  cidr_block        = "10.0.0.128/25"
+  availability_zone = "us-east-1b"
+}
+
+# Create an internet gateway to allow our cluster to access
+# the public internet. See why we need this just below.
+resource "aws_internet_gateway" "gw" {
+  vpc_id = "${aws_vpc.main.id}"
+}
+
+# Create a route from the subnet to our internet gateway so that
+# services within the subnet can access the public internet.
+# This is required so that the ECS service can pull Docker containers
+# from ECR. They're both AWS services, so why can't ECR be hosted within the subnet
+# to avoid public internet access? But apparently it has to be this way.
```

Comment:
our code is public, dont see how the containers being public would be an issue

Reply:
Well even if the code is open source you wouldn't want outsiders to have access to the code when it's running. We can secure the containers perfectly well even if they have a public IP, but having no public IP seems better if possible. I think I figured it out and in the next update the services will only have private IPs and will access the public internet using NAT.